app/services/v1/short_links/map_service.rb

```diff
@@ -24,7 +24,8 @@ def call
         session_id: available_now_session(intervention, user_intervention)&.id,
         health_clinic_id: short_link.health_clinic_id,
         multiple_fill_session_available: multiple_fill_session_available(user_intervention),
-        user_intervention_id: user_intervention&.id
+        user_intervention_id: user_intervention&.id,
+        lang: user_intervention&.language_code
```

Comment:
I think it would be better if you fetch `language_code` from the intervention instead of user_intervention. In some case, the user_intervention can be blank but you always have the intervention.

Reply:
In addition, I do not see this method in the UserIntervention model